File: Sources/RTLSCore/BatchingPolicy.swift
import Foundation

public struct BatchingPolicy: Equatable, Sendable {
    /// Upload at most this many points in one request.
    public var maxBatchSize: Int

    /// Periodically flush pending points while tracking is running.
    public var flushInterval: TimeInterval

    /// Flush when the oldest pending point is older than this (when network is available).
    public var maxBatchAge: TimeInterval

    /// When non-nil and on cellular, use this flush interval instead of `flushInterval`.
    /// Reduces upload frequency on metered connections to save battery.
    public var cellularFlushInterval: TimeInterval?

    public init(
        maxBatchSize: Int = 50,
        flushInterval: TimeInterval = 10,
        maxBatchAge: TimeInterval = 60,
        cellularFlushInterval: TimeInterval? = nil
    ) {
        self.maxBatchSize = maxBatchSize
        self.flushInterval = flushInterval
        self.maxBatchAge = maxBatchAge
        self.cellularFlushInterval = cellularFlushInterval
    }
}

File: rtls-kmp/rtls-core/src/commonMain/kotlin/com/rtls/core/BatchingPolicy.kt
package com.rtls.core

data class BatchingPolicy(
    val maxBatchSize: Int = 50,
    val flushIntervalSeconds: Long = 10L,
    val maxBatchAgeSeconds: Long = 60L,
    /** When non-null and on cellular, use this flush interval instead of [flushIntervalSeconds]. */
    val cellularFlushIntervalSeconds: Long? = null
) {
    init {
        require(maxBatchSize >= 1) { "maxBatchSize must be >= 1" }
        require(flushIntervalSeconds >= 1) { "flushIntervalSeconds must be >= 1" }
        require(maxBatchAgeSeconds >= 0) { "maxBatchAgeSeconds must be >= 0" }
    }
}
